### ISSUE TITLE
[not our bug] interrupted system calls when using object strage mounted via smb

### ISSUE THREAD

**Issue by YoSev** [minio/minio#10402]
We are trying to set up Minio with an object storage, mounted via smb.
In our case, it's the StorageBox from Hetzner.

We use the following SMB parameters via fstab on an Ubuntu 20:
//server/url /mnt/storage cifs iocharset=utf8,rw,credentials=/etc/storage-credentials.txt,uid=0,gid=0,file_mode=0660,dir_mode=0770 0       0

We tried a native binary setup and a docker based setup - both with the same results.

## Expected Behavior
Should read and write properly to the SMB Storage.

## Current Behavior
We receive a lot of (uncatched) 'interrupted system calls' whenever Minio tries to do file operations like reading the format.json etc. (during initialization etc.)
Messags like:
ERROR Unable to initialize backend: open /data/.minio.sys/format.json: interrupted system call

## Possible Solution
Adjusted parameters during SMB mount?

## Steps to Reproduce (for bugs)
1. Mount a StorageBox via SMB
2. Try to setup Minio with the Storagebox

## Context
We tried to setup a cluster of 4 Nodes but failed to initialize it, due to golang errors whenever files should be read / written.

## Regression
No

## Your Environment
Minio:RELEASE.2020-08-27T05-16-20Z, default configuration. 
Cluster and Single-Node. 
Native and Docker based.

Ubuntu 20.04. with Kernel:
Linux minio1 5.4.0-45-generic #49-Ubuntu SMP Wed Aug 26 13:38:52 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

**Comment by rluisr** [comment]
same here with same environment.

```
API: ListObjectsV2(bucket=eu-prd)
Time: 05:43:40 UTC 09/02/2020
DeploymentID: adfa856d-dc46-4fb6-b7ab-48e3d1c6a5b3
RequestID: 1630E12581FDE53B
RemoteHost: 
Host: 
UserAgent: MinIO (linux; amd64) minio-go/v7.0.5 mc/2020-08-20T00:23:01Z
Error: stat /mnt/data/minio/bucket/deleted/49584a7437.jpg: interrupted system call
       4: cmd/api-errors.go:1961:cmd.toAPIErrorCode()
       3: cmd/api-errors.go:1986:cmd.toAPIError()
       2: cmd/bucket-listobjects-handlers.go:273:cmd.objectAPIHandlers.ListObjectsV2Handler()
       1: net/http/server.go:2041:http.HandlerFunc.ServeHTTP()
```

but `/mnt/data/minio/bucket/deleted/49584a7437.jpg` is exists.

**Comment by klauspost** [comment]
Seems like a kernel bug: http://unixetc.co.uk/2020/04/06/interrupted-system-call-with-cifs-mount-on-kernel-5-3-0/

I presume a fix aws backported to `5.3.0-53` and that a later patch to `5.4.0` may have a similar fix. So maybe experiment with a newer kernel.

Either way, there is little we can do since it is an error reported by the filesystem.

**Comment by YoSev** [comment]
Makes sense.
Just gave it a quick test with the Ubuntu 18 and Kernel 4.15.0-115 and it seems to work fine - thanks.
